src/core/varlink-common.c

```diff
@@ -0,0 +1,42 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "json-util.h"
+#include "rlimit-util.h"
+#include "varlink-common.h"
+
+int rlimit_build_json(sd_json_variant **ret, const char *name, void *userdata) {
+        const struct rlimit *rl = userdata;
+        struct rlimit buf = {};
+        int r;
+
+        assert(ret);
+        assert(name);
+
+        if (!rl) {
+                const char *p;
+                int z;
+
+                /* Skip over any prefix, such as "Default" */
+                assert_se(p = strstrafter(name, "Limit"));
+
+                z = rlimit_from_string(p);
+                assert(z >= 0 && z < _RLIMIT_MAX);
+
+                r = getrlimit(z, &buf);
```

Comment:
you can drop `r` use here

Reply:
are you sure? this will lead to a error silently converted into soft=0 and hard=0. I can safe use of `r` by initialising `buf` with RLIM_INFINITY though.

Reply:
i meant in the sense of `if (getrlimit(z, &buf) < 0) {…`